src/vstd_ext/seq_lib.rs

```diff
@@ -104,13 +104,23 @@ pub proof fn seq_unequal_preserved_by_add_auto<A>(suffix: Seq<A>)
 }
 
 #[verifier(external_body)]
-pub proof fn seq_pred_false_on_all_elements_implies_empty_filter<A>(s: Seq<A>, pred: spec_fn(A) -> bool)
-    requires forall |e: A| #![auto] s.contains(e) ==> !pred(e),
-    ensures s.filter(pred).len() == 0;
+pub proof fn seq_pred_false_on_all_elements_is_equivalent_to_empty_filter<A>(s: Seq<A>, pred: spec_fn(A) -> bool)
+    ensures (forall |e: A| #![auto] s.contains(e) ==> !pred(e)) <==> s.filter(pred).len() == 0;
```

Comment:
nit: let's set a trigger instead of using auto, which makes it even harder to predict the behavior of the solver